### PR TITLE
Feature/pen test prep

### PR DESCRIPTION
**Pull request by MichaelRobinson-BJSS**

### CODE REVIEW

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Controllers/ConsentController.cs [hunk 1]
```diff
@@ -29,7 +29,10 @@ public IActionResult DismissCookieBanner()
             Response.Cookies.Append(
                 Cookies.BuyingCatalogueConsent,
                 cookieDataString,
-                new CookieOptions { Expires = DateTime.Now.Add(cookieExpiration.ConsentExpiration), });
+                new CookieOptions { Expires = DateTime.Now.Add(cookieExpiration.ConsentExpiration), HttpOnly = true, Secure = true, SameSite = SameSiteMode.Strict, });
```
Comment (bjss-chris) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Controllers/ConsentController.cs:32]: Line is getting a bit long – properties could go on separate lines.

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Controllers/ConsentController.cs [hunk 2]
```diff
@@ -29,7 +29,10 @@ public IActionResult DismissCookieBanner()
...
+
+            if (Request.GetTypedHeaders().Referer is null)
+                return Redirect("~/");
 
             return Redirect(Request.GetTypedHeaders().Referer.ToString());
```
Comment (bjss-chris) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Controllers/ConsentController.cs:37]: We can use the ternary operator here, e.g.

```c#
return Redirect(Request.GetTypedHeaders().Referer is null ? "~/" : Request.GetTypedHeaders().Referer.ToString());
```

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Program.cs [hunk 3]
```diff
@@ -49,6 +49,10 @@ public static IHostBuilder CreateHostBuilder(string[] args)
                 .ConfigureWebHostDefaults(webBuilder =>
                 {
                     webBuilder.UseStartup<Startup>();
+                    webBuilder.UseKestrel((options) =>
```
Comment (bjss-chris) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Program.cs:52]: Brackets around `options` are redundant.

**Review by bjss-chris — CHANGES_REQUESTED** [review]

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Controllers/ConsentController.cs (hunk 1 above)
Comment (MichaelRobinson-BJSS) [reply src/NHSD.GPIT.BuyingCatalogue.WebApp/Controllers/ConsentController.cs:32]: Done

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Controllers/ConsentController.cs (hunk 2 above)
Comment (MichaelRobinson-BJSS) [reply src/NHSD.GPIT.BuyingCatalogue.WebApp/Controllers/ConsentController.cs:37]: Done

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Program.cs (hunk 3 above)
Comment (MichaelRobinson-BJSS) [reply src/NHSD.GPIT.BuyingCatalogue.WebApp/Program.cs:52]: Done

**Pull request merged**